Shifty.App/Services/IProductService.cs

```diff
@@ -0,0 +1,11 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Shifty.Api.Generated.AnalogCoreV1;
+
+namespace Shifty.App.Services
+{
+    public interface IProductService
+    {
+        Task<(bool, ICollection<ProductDto>)> GetProducts();
```

Comment:
Same as for the repo

Reply:
Also, why not propagate the either to the UI? Then you would be able to show the error message it contains, if an error occured

Reply:
I had a lot of issues with propagating the Either, and when I finally succeeded for it to "catch" the Either (and not just throw another type of exception), it contained a null error (when it was in a Left state).

So I have changed it to use the TryAsync() only which I can pattern match a Success or Fail event and then grab the result or error message from there.